### PR TITLE
feat: upgrade to flutter 3.7.1

### PR DESCRIPTION
**Pull request by renancaraujo**
## Description

feat: upgrade to flutter 3.7.0

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]
LGTM once CI is passing

File: .github/workflows/my_plugin.yaml [hunk 1]
```diff
@@ -21,11 +21,12 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.1
       working_directory: src/my_plugin/my_plugin
 
   android:
     runs-on: macos-latest
+    if: false # TODO: Integration tests are flaky since Flutter 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin.yaml:29]: Thoughts on leaving these enabled but just removing them from the required status checks in GitHub? That way we can keep an eye on whether the issue is resolved.

**Comment by renancaraujo** [comment]
Added an issue to track the failing jobs: https://github.com/VeryGoodOpenSource/very_good_flutter_plugin/issues/79

File: .github/workflows/my_plugin_android.yaml [hunk 2]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_android.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: .github/workflows/my_plugin_ios.yaml [hunk 3]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_ios.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: .github/workflows/my_plugin_linux.yaml [hunk 4]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_linux.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: .github/workflows/my_plugin_macos.yaml [hunk 5]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_macos.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: .github/workflows/my_plugin_platform_interface.yaml [hunk 6]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_platform_interface.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: .github/workflows/my_plugin_web.yaml [hunk 7]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_web.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: .github/workflows/my_plugin_windows.yaml [hunk 8]
```diff
@@ -21,5 +21,5 @@ jobs:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/flutter_package.yml@v1
     with:
       flutter_channel: stable
-      flutter_version: 3.3.2
+      flutter_version: 3.7.0
```
Comment (felangel) [inline .github/workflows/my_plugin_windows.yaml:24]:
```suggestion
      flutter_version: 3.7.1
```

File: patches/android/workflow_pana.patch [hunk 9]
```diff
@@ -4,7 +4,7 @@ index 1f38a07..9116a1b 100644
 +++ b/.github/workflows/my_plugin_android.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/android/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/ios/workflow_pana.patch [hunk 10]
```diff
@@ -4,7 +4,7 @@ index 554a9b0..fb43a09 100644
 +++ b/.github/workflows/my_plugin_ios.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/ios/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/linux/workflow_pana.patch [hunk 11]
```diff
@@ -4,7 +4,7 @@ index c9a46df..c08e551 100644
 +++ b/.github/workflows/my_plugin_linux.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/linux/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/macos/workflow_pana.patch [hunk 12]
```diff
@@ -4,7 +4,7 @@ index 6e13220..5e20190 100644
 +++ b/.github/workflows/my_plugin_macos.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/macos/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/platform_interface/workflow_pana.patch [hunk 13]
```diff
@@ -4,7 +4,7 @@ index b336f54..043beb6 100644
 +++ b/.github/workflows/my_plugin_platform_interface.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/platform_interface/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/plugin/workflow_pana.patch [hunk 14]
```diff
@@ -4,7 +4,7 @@ index 6f98e99..72bea51 100644
 +++ b/.github/workflows/my_plugin.yaml
 @@ -19,6 +19,10 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/plugin/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/web/workflow_pana.patch [hunk 15]
```diff
@@ -4,7 +4,7 @@ index 157bfdc..91934d4 100644
 +++ b/.github/workflows/my_plugin_web.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/web/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: patches/windows/workflow_pana.patch [hunk 16]
```diff
@@ -4,7 +4,7 @@ index 73ac9c9..ede6122 100644
 +++ b/.github/workflows/my_plugin_windows.yaml
 @@ -19,3 +19,7 @@ jobs:
        flutter_channel: stable
-       flutter_version: 3.3.2
+       flutter_version: 3.7.0
```
Comment (felangel) [inline patches/windows/workflow_pana.patch:7]:
```suggestion
       flutter_version: 3.7.1
```

File: .github/workflows/my_plugin.yaml (hunk 1 above)
Comment (renancaraujo) [reply .github/workflows/my_plugin.yaml]: For the record: We decided to keep this line as we cant disable specific workflows on gh settings

**Review by erickzanardo — APPROVED** [review]

File: .github/CODEOWNERS [hunk 17]
```diff
@@ -1,3 +1,3 @@
 # Every request must be reviewed and accepted by:
 
-*	@felangel
\ No newline at end of file
+*	@felangel @renancaraujo @wolfenrain
```
Comment (felangel) [inline .github/CODEOWNERS:3]: 💯

**Review by felangel — APPROVED** [review]

**Pull request merged**